src/ReaderImpl.cpp

```diff
@@ -627,7 +627,20 @@ namespace e57
       const StructureNode proto( points.prototype() );
 
       data3DHeader.guid = StringNode( scan.get( "guid" ) ).value();
+
+#ifdef E57_32_BIT
+      // If we exceed the size_t max, only process the max (4,294,967,295 points).
+      if ( points.childCount() > static_cast<int64_t>( std::numeric_limits<size_t>::max() ) )
+      {
+         data3DHeader.pointCount = static_cast<int64_t>( std::numeric_limits<size_t>::max() );
+      }
+      else
+      {
+         data3DHeader.pointCount = static_cast<size_t>( points.childCount() );
+      }
```

Comment:
Doesn't this throw a warning as one time you assign a signed type (`int64_t`) to `pointCount` and one time a unsigned type (`size_t`)?

I don't see the type of `pointCount` here, but you can simplify the code to sth like this:
```suggestion
      data3DHeader.pointCount = std::min<int64_t>( points.childCount(), std::numeric_limits<size_t>::max() );
```
As `int64_t` is passed as template parameter, both parameter will be cast implicit to this type.

Reply:
Yeah - that second cast is incorrect. `data3DHeader.pointCount` is `size_t`. This is a unsigned-to-signed conversion (to size_t), so I don't think std::min can be used here.

I added some output here (see below) so I need the "if" anyways.